lab-remil/lib/storage.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('fs'), { suffix: 'Prom'} );
+
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item) {
+  if (!schemaName) return Promise.reject(new Error('expected schema'));
+  if (!item) return Promise.reject(new Error('expected item'));
+
+  fs.writeFileProm(`${__dirname}/../data/${schemaName}/${item.id}.json`, JSON.stringify(item))
+  .then(() => item)
+  .catch( err => Promise.reject(err));
+};
+
+exports.fetchItem = function(schemaName, id) {
+  if (!schemaName) return Promise.reject(new Error('expected schema'));
+  if (!id) return Promise.reject(new Error('expected id'));
+
+  return fs.readFileProm(`${__dirname}/../data/${schemaName}/${id}.json`)
+  .then( data => {
+    try {
+      let item = JSON.parse(data.toString());
+      return item;
+    } catch (err) {
+      return Promise.reject(err);
+    }
+  })
+  .catch( err => Promise.reject(err));
+};
+
+exports.deleteItem = function(schemaName, id) {
+  if (!schemaName) return Promise.reject(new Error('expected schema'));
+  if (!id) return Promise.reject(new Error('expected id'));
+
+  return fs.unlinkProm(`${__dirname}/../data/${schemaName}/${id}.json`);
```

Comment:
because `fs.unlinkProm` returns a promise you might want to consider catching an error here in case the `fs.unlinkProm` fails. (i.e. `.catch( err => Promise.reject(err));`)